Backend/Tournament_User/TournamentUserMVSC/middlewares/errorHandler.js

```diff
@@ -0,0 +1,23 @@
+// Error-handling middleware for Express
+const errorHandler = (err, req, res, next) => {
+    console.error(err.stack);  // Log the error for debugging purposes
+
+    // Handle custom database-related errors (like MySQL SIGNAL errors)
+    if (err.sqlState === '45000') {
+        return res.status(400).json({ message: err.message });
```

Comment:
i dont think itll reach here according to the code above? but yeah pls centralise all error handling